atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/NextTableToSweepProvider.java

```diff
@@ -96,6 +97,12 @@ public static NextTableToSweepProvider create(KeyValueService kvs, LockService l
         List<TableReference> priorityTableRefs = overrideConfig.priorityTablesAsList().stream()
                 .map(TableReference::createFromFullyQualifiedName)
                 .collect(Collectors.toList());
+
+        // If there are multiple priority tables, we don't want to consistently use the same ordering.
+        // It is true that this operation is O(list length) while an O(min(list length, cluster size)) algorithm
+        // exists, but the priority table reference list is expected to be small.
+        Collections.shuffle(priorityTableRefs);
+
         return attemptToChooseTableFromList(priorityTableRefs, "it is on the sweep priority list");
```

Comment:
Detail: I would rename this method to "attemptToChooseTableFromPrioritisedList" to make this sort of bug a little clearer.